tests/dialects/test_builtin.py

```diff
@@ -45,3 +49,90 @@ def test_DenseArrayBase_verifier_failure():
         DenseArrayBase([i32, ArrayAttr.from_list([FloatData(0.0)])])
     assert err.value.args[0] == ("dense array of integer element type "
                                  "should only contain integers")
+
+
+def test_vector_rank_constraint_verify():
+    vector_type = VectorType.from_element_type_and_shape(i32, [1, 2])
+    constraint = VectorRankConstraint(2)
+
+    constraint.verify(vector_type)
+
+
+def test_vector_rank_constraint_rank_mismatch():
+    vector_type = VectorType.from_element_type_and_shape(i32, [1, 2])
+    constraint = VectorRankConstraint(3)
+
+    with pytest.raises(VerifyException) as e:
+        constraint.verify(vector_type)
+    assert e.value.args[0] == "Expected vector rank to be 3, got 2."
+
+
+def test_vector_rank_constraint_attr_mismatch():
+    memref_type = MemRefType.from_element_type_and_shape(i32, [1, 2])
+    constraint = VectorRankConstraint(3)
+
+    with pytest.raises(VerifyException) as e:
+        constraint.verify(memref_type)
+    assert e.value.args[
+        0] == "!memref<[1 : !index, 2 : !index], !i32> should be of type VectorType."
```

Comment:
(this is something more general, not relate to this PR) IMO, it would look better if we could print the attributes without the `!` in error messages.

Reply:
thoughts? @math-fehr @superlopuh